models/BaseEntity.cfc

```diff
@@ -1165,10 +1183,36 @@ component accessors="true" {
 	 * @return  quick.models.BaseEntity
 	 */
 	private any function loadEntity( required struct data ) {
-		return newEntity()
-			.assignAttributesData( arguments.data )
-			.assignOriginalAttributes( arguments.data )
-			.markLoaded();
+		if (
+			variables._loadChildren
+			&&
+			variables._meta.localMetadata.keyExists( "discriminatorColumn" )
+			&&
+			application.quickMeta.discriminators.keyExists( variables._meta.table )
+			&&
+			data.keyExists( variables._meta.localMetadata.discriminatorColumn )
+			&&
+			application.quickMeta.discriminators[ variables._meta.table ].keyExists(
+				data[ variables._meta.localMetadata.discriminatorColumn ]
+			)
+		) {
+			var childClass = variables._wirebox.getInstance(
+				application.quickMeta.discriminators[ variables._meta.table ][
+					data[ variables._meta.localMetadata.discriminatorColumn ]
+				].mapping
+			);
+
+			keyNames().each( function( key, i ) {
```

Comment:
Can you help me understand what this section is doing?  I'm feeling a bit confused.  It feels like it would override ids, so I must be missing something.

Reply:
The key for the child class is the join column, so this is taking the parent keys and assigning the join column names in to the data structure used to hydrate the child.  Without this code, the call to `hydrate` throws and exception because the keys aren't present.